### DISCUSSION TITLE
Using Workbench in Docker

### DISCUSSION THREAD

**Discussion by hardwayh** [jamesdolezal/slideflow#240] [General]
Hi James,

Thanks for sharing your code and documentation!  I am using the Docker image (very helpful!) but am unable to run workbench in it -- do you have other details on settings?  I have tried setting the DISPLAY variable to suggestions found online (:0), but still get an error that tkinter cannot connect to display.  Thanks so much for your help!

Heather

<img width="834" alt="Screenshot 2023-03-29 at 4 07 42 PM" src="https://user-images.githubusercontent.com/126614519/228668041-67971ebf-fdb7-4c6b-9d28-283445cdd82f.png">

<img width="826" alt="Screenshot 2023-03-29 at 4 10 12 PM" src="https://user-images.githubusercontent.com/126614519/228668314-74c3e47b-8bbc-431e-a17f-4076ceff20c4.png">

**Comment by jamesdolezal** [comment]
Hi Heather,

We actually haven't done much testing of Workbench on Docker - our Docker testing protocol has so far included all other functions in Slideflow apart from Workbench. It looks like Tkinter and GLFW (the OpenGL framework we use for GUI rendering) are both potentially compatible with Docker, so I will investigate and get back with you. I've created an issue (#241) that we can use for tracking status and progress.

Also as a heads up - Workbench is getting a major overhaul in 1.6.0 and will be renamed Slideflow Studio. It includes a number of new additional functions (including ROI creation and editing), a nicer interface, expanded documentation, and some additional optimizations. We'll be sure to include details about Docker compatibility in the documentation when the update goes live. We're anticipating release in about 2 weeks.

**Comment by hardwayh** [comment]
Hi James,

Thanks for the quick response and looking into my question!  I’m new to digital path but so far, love your tool, so look forward to the new release!  Thanks again!

Heather

From: James Dolezal ***@***.***>
Date: Wednesday, March 29, 2023 at 5:04 PM
To: jamesdolezal/slideflow ***@***.***>
Cc: Hardway, Heather D., Ph.D. ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [jamesdolezal/slideflow] Using Workbench in Docker (Discussion #240)

Hi Heather,

We actually haven't done much testing of Workbench on Docker - our Docker testing protocol has so far included all other functions in Slideflow apart from Workbench. It looks like Tkinter and GLFW (the OpenGL framework we use for GUI rendering) are both potentially compatible with Docker, so I will investigate and get back with you. I've created an issue (#241<https://nam12.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fjamesdolezal%2Fslideflow%2Fissues%2F241&data=05%7C01%7CHardway.Heather%40mayo.edu%7C49b24aba4fd043a5e12708db30a19194%7Ca25fff9c3f634fb29a8ad9bdd0321f9a%7C0%7C0%7C638157242723143064%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=TFV7YEQFUiu%2FrSybHs0lW5YTxlits%2F%2BO%2FknbZU4ybZ4%3D&reserved=0>) that we can use for tracking status and progress.

Also as a heads up - Workbench is getting a major overhaul in 1.6.0 and will be renamed Slideflow Studio. It includes a number of new additional functions (including ROI creation and editing), a nicer interface, expanded documentation, and some additional optimizations. We'll be sure to include details about Docker compatibility in the documentation when the update goes live. We're anticipating release in about 2 weeks.
